pyteal/ast/abi/tuple.py

```diff
@@ -341,17 +341,18 @@ def set(self, *values):
         if len(values) == 1 and isinstance(values[0], ComputedValue):
             return self._set_with_computed_type(values[0])
 
-        for value in values:
-            if not isinstance(value, BaseType):
-                raise TealInputError(f"Expected BaseType, got {value}")
-
         myTypes = self.type_spec().value_type_specs()
         if len(myTypes) != len(values):
             raise TealInputError(
                 f"Incorrect length for values. Expected {len(myTypes)}, got {len(values)}"
             )
-        if not all(myTypes[i] == values[i].type_spec() for i in range(len(myTypes))):
-            raise TealInputError("Input values do not match type")
+        for index, (value, myType) in enumerate(zip(values, myTypes)):
+            if not isinstance(value, BaseType):
+                raise TealInputError(f"Expected BaseType, got {value}")
+            if not myType == value.type_spec():
+                raise TealInputError(
+                    f"Input values do not match type at {index}: {value.type_spec()} != {myType}"
```

Comment:
nit: minor err msg phrasing
```suggestion
                    f"Input values do not match type at {index=}: {value.type_spec()} != {myType}"
```